numba/core/compiler.py

```diff
@@ -642,6 +642,7 @@ def define_untyped_pipeline(state, name='untyped'):
 
         pm.add_pass(FindLiterallyCalls, "find literally calls")
         pm.add_pass(LiteralUnroll, "handles literal_unroll")
+        pm.add_pass(RewriteSemanticConstants, "rewrite semantic constants")
```

Comment:
Apologies, I meant to say to put this after this pass:
https://github.com/numba/numba/blob/553a7d7cc757107ac1491763d3dabfaddab299e4/numba/core/untyped_passes.py#L1657
i.e. it only needs to run if the `literal_unroll` subpipeline is also being run.